apps/andi/lib/andi_web/live/submit_live_view/upload_data_dictionary.ex

```diff
@@ -0,0 +1,248 @@
+  alias AndiWeb.ErrorHelpers
+  alias AndiWeb.InputSchemas.DatasetLinkFormSchema
+  alias AndiWeb.Helpers.FormTools
+  alias ExAws.S3
+
+  plug Tesla.Middleware.Timeout, timeout: 20_000
+
+  @bucket_path "samples/"
+
+  getter(:hosted_bucket, generic: true)
+
+  def mount(_, %{"dataset" => dataset}, socket) do
+    new_upload_data_dictionary_changeset = AndiWeb.InputSchemas.DatasetLinkFormSchema.changeset_from_andi_dataset(dataset)
+
+    send(socket.parent_pid, {:update_dataset_link_status, new_upload_data_dictionary_changeset.valid?})
+
+    AndiWeb.Endpoint.subscribe("toggle-visibility")
+    AndiWeb.Endpoint.subscribe("form-save")
+
+    {:ok,
+     assign(socket,
+       changeset: new_upload_data_dictionary_changeset,
+       visibility: "collapsed",
+       validation_status: "collapsed",
+       dataset_id: dataset.id,
+       loading_schema: false
+     )}
+  end
+
+  def render(assigns) do
+    action =
+      case assigns.visibility do
+        "collapsed" -> "EDIT"
+        "expanded" -> "MINIMIZE"
+      end
+
+    loader_visibility =
+      case assigns.loading_schema do
+        true -> "loading"
+        false -> "hidden"
+      end
+
+    ~L"""
+    <div id="upload-sample-dataset-form" class="form-component">
+      <div class="component-header" phx-click="toggle-component-visibility">
+        <div class="section-number">
+          <h3 class="component-number component-number--<%= @validation_status %> component-number--<%= @visibility %>">2</h3>
+          <div class="component-number-status--<%= @validation_status %>"></div>
+        </div>
+
+        <div class="component-title full-width">
+          <h2 class="component-title-text component-title-text--<%= @visibility %>">Upload Dataset Sample</h2>
+          <div class="component-title-action">
+            <div class="component-title-action-text--<%= @visibility %>"><%= action %></div>
+            <div class="component-title-icon--<%= @visibility %>"></div>
+          </div>
+        </div>
+      </div>
+
+      <div class="form-section">
+        <%= f = form_for @changeset, "#", [phx_change: :validate, as: :form_data, multipart: true] %>
+       
+          <div class="component-edit-section--<%= @visibility %>">
+            <div class="url-form-header">
+              <h4>Dataset sample may not contain any Personally Identifiable Information (PII). If the data is found to contain PII, it will be rejected.</h4>
+            </div>
+            <div class="upload-section">
+
+              <div class="upload-data-dictionary-form__file-upload">
+                <div class="file-input-button">
+                  <%= label(f, :sample_dataset, "Select File", class: "file-upload-label") %>
+                  <%= file_input(f, :sample_dataset, phx_hook: "readFile", accept: "text/csv, application/json") %>
+                </div>
+                <div class="sample-file-display">
+                  <%= hidden_input(f, :datasetLink) %>
```

Comment:
I don't think you need a hidden input if you are already referencing the value on the line below. maybe I'm wrong but what are you using this for?

Reply:
I use this so I can display the file name that was uploaded. I had to override the styling of the upload button so it no longer does this by default